packages/components/src/border-box-control/test/index.tsx

```diff
@@ -260,27 +273,34 @@ describe( 'BorderBoxControl', () => {
 
 			const widthInputs = screen.getAllByRole( 'spinbutton', {
 				name: 'Border width',
-			} );
+			} ) as HTMLInputElement[];
 			expect( widthInputs[ 0 ].value ).toBe( '1' ); // Top.
 			expect( widthInputs[ 1 ].value ).toBe( '1' ); // Left.
 			expect( widthInputs[ 2 ].value ).toBe( '1' ); // Right.
 			expect( widthInputs[ 3 ].value ).toBe( '1' ); // Bottom.
 		} );
 
-		it( 'should omit style options when requested', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
+		// We're expecting to have 4 color buttons by default.
+		const colorButtonIndexes = [ ...Array( 4 ).keys() ];
 
-			render( <BorderBoxControl { ...props } enableStyle={ false } /> );
+		it.each( colorButtonIndexes )(
+			'should omit style options when color button %s is pressed',
+			async ( colorButtonIndex ) => {
+				const user = userEvent.setup();
 
-			await user.click(
-				screen.getByRole( 'button', { name: 'Unlink sides' } )
-			);
+				render(
+					<BorderBoxControl { ...props } enableStyle={ false } />
+				);
+
+				await user.click(
+					screen.getByRole( 'button', { name: 'Unlink sides' } )
+				);
+
+				const colorButtons =
+					screen.getAllByLabelText( colorPickerRegex );
 
-			const colorButtons = screen.getAllByLabelText( colorPickerRegex );
+				await user.click( colorButtons[ colorButtonIndex ] );
 
-			function assertStyleOptionsMissing() {
 				const styleLabel = screen.queryByText( 'Style' );
 				const solidButton = screen.queryByRole( 'button', {
```

Comment:
We could be able to shorten this to a single query, where:

```
const solidButton = screen.queryByRole( 'button', {
	name: 'Solid',
} );
const dashedButton = screen.queryByRole( 'button', {
	name: 'Dashed',
} );
const dottedButton = screen.queryByRole( 'button', {
	name: 'Dotted',
} );

expect( solidButton ).not.toBeInTheDocument();
expect( dashedButton ).not.toBeInTheDocument();
expect( dottedButton ).not.toBeInTheDocument();

```

to:

```
const styleButton = screen.queryByRole( 'button', {
	name: /(Solid)|(Dashed)|(Dotted)/,
} );

expect( styleButton ).not.toBeInTheDocument();
```

This is totally optional since it's a suggestion on pre-existing code.

Reply:
Clever! Applied in https://github.com/WordPress/gutenberg/pull/47755/commits/d7b188b251743e11181e81cf551442f57f5c9c82

Reply:
Nice. You might also want to do it for the other `should omit style options when requested` test that has the same assertions.